lib/improver/ensemble_calibration/ensemble_calibration.py

```diff
@@ -1101,318 +1100,3 @@ def format_calibration_method(calibration_method):
-            percentiles):
-        """
-        Function returning percentiles based on the supplied
-        mean and variance. The percentiles are created by assuming a
-        Gaussian distribution and calculating the value of the phenomenon at
-        specific points within the distribution.
-
-        Parameters
-        ----------
-        calibrated_forecast_predictor : cube
-            Predictor for the calibrated forecast i.e. the mean.
-        calibrated_forecast_variance : cube
-            Variance for the calibrated forecast.
-        percentiles : List
-            Percentiles at which to calculate the value of the phenomenon at.
-
-        Returns
-        -------
-        percentile_cube : Iris cube
-            Cube containing the values for the phenomenon at each of the
-            percentiles requested.
-
-        """
-        if not calibrated_forecast_predictor.coord_dims("time"):
-            calibrated_forecast_predictor = iris.util.new_axis(
-                calibrated_forecast_predictor, "time")
-        if not calibrated_forecast_variance.coord_dims("time"):
-            calibrated_forecast_variance = iris.util.new_axis(
-                calibrated_forecast_variance, "time")
-
-        calibrated_forecast_predictor_data = (
-            calibrated_forecast_predictor.data.flatten())
-        calibrated_forecast_variance_data = (
-            calibrated_forecast_variance.data.flatten())
-
-        result = np.zeros((calibrated_forecast_predictor_data.shape[0],
-                           len(percentiles)))
-
-        # Loop over percentiles, and use a normal distribution with the mean
-        # and variance to calculate the values at each percentile.
-        for index, percentile in enumerate(percentiles):
-            percentile_list = np.repeat(
-                percentile, len(calibrated_forecast_predictor_data))
-            result[:, index] = norm.ppf(
-                percentile_list, loc=calibrated_forecast_predictor_data,
-                scale=np.sqrt(calibrated_forecast_variance_data))
-            # If percent point function (PPF) returns NaNs, fill in
-            # mean instead of NaN values. NaN will only be generated if the
-            # variance is zero. Therefore, if the variance is zero, the mean
-            # value is used for all gridpoints with a NaN.
-            if np.any(calibrated_forecast_variance_data == 0):
-                nan_index = np.argwhere(np.isnan(result[:, index]))
-                result[nan_index, index] = (
-                    calibrated_forecast_predictor_data[nan_index])
-            if np.any(np.isnan(result)):
-                msg = ("NaNs are present within the result for the {} "
-                       "percentile. Unable to calculate the percent point "
-                       "function.")
-                raise ValueError(msg)
-
-        result = result.T
-
-        t_coord = calibrated_forecast_predictor.coord("time")
```

Comment:
This code has been modified to make it more generic in the new version. No changes to this function above this point.